### PR TITLE
Rush processing endpoint

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/jag-efiling-api.yaml [hunk 1]
```diff
@@ -230,6 +230,25 @@ paths:
...
+      requestBody:
+        description: filing submit parameters
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Rush"
+      responses:
+        "201":
+          description: Rush created
```
Comment (akroon3r) [inline src/backend/efiling-api/jag-efiling-api.yaml:251]: failure case?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 2]
```diff
@@ -430,4 +434,35 @@ private ResponseEntity storeDocuments(SubmissionKey submissionKey, List<Multipar
         return ResponseEntity.ok(new UploadSubmissionDocumentsResponse().submissionId(submissionKey.getSubmissionId()).received(new BigDecimal(files.size())));
     }
 
+    @Override
+    @RolesAllowed("efiling-user")
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:438]: do we need a role like early adopters to roll this out in prod?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 3]
```diff
@@ -430,4 +434,35 @@ private ResponseEntity storeDocuments(SubmissionKey submissionKey, List<Multipar
...
+        if(!universalId.isPresent()) {
+
+            logger.error(UNIVERSAL_ID_IS_REQUIRED);
+            throw new InvalidUniversalException(INVALID_UNIVERSAL_ID);
+
+        }
+
+        SubmissionKey submissionKey = new SubmissionKey(universalId.get(), xTransactionId, submissionId);
+
+        logger.info("attempting to add rush processing to efiling package for transaction [{}]", xTransactionId);
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:452]:
```suggestion
        logger.info("Attempting to add rush processing to efiling package for transaction [{}]", xTransactionId);
```

Comment (akroon3r) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:452]: is the transaction ID too much for prod logging?

**Review by akroon3r — COMMENTED** [review]
Couple of questions - sorry.

File: src/backend/efiling-api/jag-efiling-api.yaml (hunk 1 above)
Comment (TayGov) [reply src/backend/efiling-api/jag-efiling-api.yaml:251]: Handled in code but re-added. We are pretty inconsistent with setting these.

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java (hunk 2 above)
Comment (TayGov) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:438]: No

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java (hunk 3 above)
Comment (TayGov) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:452]: No and we are logging it everywhere. It doesn't get stored in the database

Comment (TayGov) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:452]: Also that comment is consistent with others

**Review by akroon3r — APPROVED** [review]
🐦

**Pull request merged**